[ios] Don't pause camera animations if zoom/pitch/rotate/scroll are disabled

captainbarbosa opened the pull request:
Closes https://github.com/mapbox/mapbox-gl-native/issues/11415.

Currently when a camera animation is triggered programmatically, if the `zoomEnabled` / `pitchEnabled` / `scrollEnabled` / `rotateEnabled` are all set to `NO`/`FALSE`, user interaction causes the transition to stop. This PR reverses that decision, so that a programmatic camera transition will continue if the above four properties are set to `NO`/`FALSE`

friedbunny commented on platform/ios/src/MGLMapView.mm:
```diff
@@ -1376,6 +1376,11 @@ - (void)touchesBegan:(__unused NSSet<UITouch *> *)touches withEvent:(__unused UI
+    
+    if (!self.zoomEnabled && !self.pitchEnabled && !self.rotateEnabled && !self.scrollEnabled) {
+        return;
+    };
```
Can you explain your decision to place this here (and not as the first thing in this method)?

captainbarbosa replied:
No specific reason 😅 I didn't realize it at the time, but it makes sense to move it up so that we can return earlier if need be.

friedbunny commented:
What’s the MapKit behavior in this case, re: camera and user tracking?

captainbarbosa commented:
@friedbunny see https://github.com/mapbox/mapbox-gl-native/issues/11415#issuecomment-414792187 to check out my comparison from researching the behavior on other platforms:

>I compared this behavior to some other maps SDKs, mainly Google Maps and MapKit. When the related is*Enabled property is set to false in MapKit, a user is still able to stop a programmatic camera transition when interacting with the display. The Google Maps SDK however, doesn't allow a user interaction to stop a camera transition when the associated *Gestures properties are set to false.

I went ahead and addressed your review and included the changelog entry as well.

julianrex commented:
@tobrun what's the similar behavior on Android?

tobrun commented:
> @tobrun what's the similar behavior on Android?

We don't interrupt animations when user input is disabled

julianrex commented:
@captainbarbosa @friedbunny Given that MapKit and Google differ, I think we should be consistent with our Android SDK.
